execution/executor/src/chunk_executor.rs

```diff
@@ -261,6 +256,11 @@ impl<V: VMExecutor> ChunkExecutorTrait for ChunkExecutor<V> {
         )?;
         self.commit_chunk()
     }
+
+    fn reset(&self) -> Result<()> {
```

Comment:
I always hated the `reset()` interface, and was willing to kill it after the legacy interfaces `x_and_y()` gets removed together with state sync v1. Is it possible on the call site to just recreate the `ChunkExecutor`? 

(Doesn't feel too strong either, probably not worth a huge refactor.)